pkg/controllers/nodeclass/validation.go

```diff
@@ -191,74 +201,37 @@ func (v *Validation) validateRunInstancesAuthorization(
 	nodeClaim *karpv1.NodeClaim,
 	tags map[string]string,
 ) (reason string, requeue bool, err error) {
-	// NOTE: Since we've already validated the status conditions are true, these should never occur
-	if len(nodeClass.Status.AMIs) == 0 {
```

Comment:
Just to verify, we are dropping this since we have done the check at the start of the reconcile, correct?

Reply:
It's already handled by the `Resolve()` method on the amiResolver